src/main/java/com/indeed/authorization/client/validators/IndeedAccessTokenValidator.java

```diff
@@ -0,0 +1,81 @@
+package com.indeed.authorization.client.validators;
+
+import com.indeed.authorization.client.claims.IndeedAccessTokenClaimsSet;
+import com.indeed.authorization.client.tokens.IndeedAccessToken;
+import com.nimbusds.jose.jwk.source.RemoteJWKSet;
+import com.nimbusds.jose.proc.JWKSecurityContext;
+import com.nimbusds.jose.proc.JWSKeySelector;
+import com.nimbusds.jose.proc.JWSVerificationKeySelector;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.BadJWTException;
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import com.nimbusds.oauth2.sdk.id.Issuer;
+import com.nimbusds.openid.connect.sdk.validators.AbstractJWTValidator;
+
+import java.net.URL;
+
+public class IndeedAccessTokenValidator extends AbstractJWTValidator {
+    private IndeedAccessTokenValidator(
+            final Issuer expectedIssuer,
+            final ClientID clientID,
+            final JWSKeySelector jwsKeySelector) {
+        super(expectedIssuer, clientID, jwsKeySelector, null);
+    }
+
+    /**
+     * Validates the access token against the expected scopes.
+     *
+     * @param accessToken The generated access token. Not Null.
+     * @param expectedScopes The expected scopes to the access token. Not Null.
+     * @return {@link IndeedAccessTokenClaimsSet}
+     * @throws BadJWTException
+     * @throws ParseException
+     */
+    public IndeedAccessTokenClaimsSet validate(
+            final IndeedAccessToken accessToken, final String[] expectedScopes)
+            throws BadJWTException, ParseException {
+        if (accessToken == null) {
+            throw new IllegalArgumentException("The access token issuer must not be null");
+        }
+
+        final JWTClaimsSet claimsSet;
+
+        try {
+            claimsSet = accessToken.getJWTClaimsSet();
+        } catch (final java.text.ParseException e) {
+            throw new BadJWTException(e.getMessage(), e);
+        }
+
+        final IndeedAccessTokenClaimsVerifier verifier =
+                new IndeedAccessTokenClaimsVerifier(
+                        this.getExpectedIssuer(),
+                        this.getClientID(),
+                        expectedScopes,
+                        getMaxClockSkew());
+        verifier.verify(claimsSet, null);
+        return new IndeedAccessTokenClaimsSet(claimsSet);
+    }
+
+    /**
+     * Create an {@link IndeedAccessTokenValidator} Object.
+     *
+     * @param issuer Indeed's issuer (ie. <a
+     *     href="https://secure.indeed.com">https://secure.indeed.com</a>
+     * @param jwkSetUrl The key selector for JWS verification, null if unsecured (plain) tokens are
```

Comment:
If `null` is simply not acceptable (L72~73), the javadoc should simply say must not be null.

Also, similar with the issuer, I don't expect it to be configured as a different url than https://secure.indeed.com/.well-known/keys. How about simply hardcode them?

Reply:
This has been addressed. Similar to my reply to the issuer, since internal projects are using the library, the host may be different.